custom/lib/Auth.php

```diff
@@ -48,23 +53,19 @@ public function __construct(private string $appId, private Configuration $config
      *
      * @return string User ID of the Passage user
      * @throws InvalidArgumentException JWT format is invalid
-     * @throws UnexpectedValueException Could not retrieve sub claim from token
+     * @throws UnexpectedValueException Could not validate aud claim or retrieve sub claim from token
      */
     public function validateJwt(string $jwt): string
     {
-        $jwtSegments = explode('.', $jwt);
-        if (count($jwtSegments) !== 3) {
-            throw new InvalidArgumentException('Invalid JWT format');
-        }
+        $decodedToken = JWT::decode($jwt, $this->jwks);
 
-        $decodedHeader = JWT::urlsafeB64Decode($jwtSegments[0]);
-        $header = json_decode($decodedHeader);
+        $aud = (array) $decodedToken->aud;
+        $expectedAud = $this->app->getHosted() ? $this->appId : $this->app->getAuthOrigin();
 
-        if (!$header->kid) {
-            throw new InvalidArgumentException('Missing kid in token');
+        if (!in_array($expectedAud, $aud)) {
+            throw new UnexpectedValueException('JWT audience does not match');
         }
```

Comment:
this is already checked by `JWT::decode` [here](https://github.com/firebase/php-jwt/blob/29fa2ce9e0582cd397711eec1e80c05ce20fabca/src/JWT.php#L112-L115), [here](https://github.com/firebase/php-jwt/blob/29fa2ce9e0582cd397711eec1e80c05ce20fabca/src/JWT.php#L138), and [here](https://github.com/firebase/php-jwt/blob/29fa2ce9e0582cd397711eec1e80c05ce20fabca/src/JWT.php#L467-L469)